packages/babel-plugin-transform-es2015-block-scoping/src/index.js

```diff
@@ -488,17 +488,31 @@ class BlockScoping {
       }
     }
 
+    const addDeclarationsFromChild = (path, node) => {
+      node = node || path.node;
+      if (t.isClassDeclaration(node) || t.isFunctionDeclaration(node) || isBlockScoped(node)) {
+        if (isBlockScoped(node)) {
+          convertBlockScopedToVar(path, node, block, this.scope);
+        }
+        declarators = declarators.concat(node.declarations || node);
```

Comment:
👍 Did something similar in d276d57. CI is being fussy though: https://travis-ci.org/babel/babel/jobs/165258111#L161

Grumble grumble. Let me try rebasing to latest master.